### PR TITLE
app-backup/btrbk: add versions 0.23.2 and 9999

### PR DESCRIPTION
**Pull request by candrews**
btrbk is a backup tool for btrfs subvolumes, taking advantage of btrfs specific capabilities to create atomic snapshots and transfer them incrementally to backup locations via ssh.

https://bugs.gentoo.org/show_bug.cgi?id=564780

### CODE REVIEW

**Comment by candrews** [comment]
@gentoo/proxy-maint How's this look?

File: app-backup/btrbk/btrbk-0.23.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	SRC_URI=""
+	KEYWORDS=""
+else
+	SRC_URI="https://digint.ch/download/btrbk/releases/${P}.tar.xz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Tool for creating snapshots and remote backups of btrfs subvolumes"
+HOMEPAGE="https://digint.ch/btrbk/"
+LICENSE="GPL-3"
```
Comment (gktrk) [inline app-backup/btrbk/btrbk-0.23.2.ebuild]: The correct license is GPL-3+

File: app-backup/btrbk/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+    <maintainer type="person">
+        <email>candrews@integralblue.com</email>
+        <name>Craig Andrews</name>
+        <description>Proxied maintainer; set to assignee in all bugs</description>
```
Comment (gktrk) [inline app-backup/btrbk/metadata.xml]: &lt;description&gt; isn't used anymore. Per GLEP67, the first maintainer is the default bug assignee.

File: app-backup/btrbk/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    </maintainer>
+    <maintainer type="project">
+        <email>proxy-maint@gentoo.org</email>
+        <name>Proxy Maintainers</name>
+    </maintainer>
+    <use>
```
Comment (gktrk) [inline app-backup/btrbk/metadata.xml]: If there are no USE flags &lt;use/&gt; block isn't needed.

**Comment by gktrk** [comment]
@gentoo/proxy-maint

**Comment by gktrk** [comment]
Is there a reason why we prefer manual installation over something like:
`emake DESTDIR="${D}" DOCDIR="${D}/usr/share/${PF}" install`

**Comment by candrews** [comment]
@gktrk good point, this change really reduces the number of lines / complexity of the ebuild and improves maintainability.
Here's what I'm doing now:
```
src_install() {
        emake DESTDIR="${D}" DOCDIR="/usr/share/doc/${PF}" install
}
```

I compared the output of `equery files btrbk` before and after this change. They're identical except for 2 small differences:
* The ChangeLog isn't installed by the Makefile - I don't think this is worth doing in the ebuild, but I asked upstream to make this change: https://github.com/digint/btrbk/pull/95
* The ebuild was installing the scripts in /usr/share/btrbk instead of /usr/share/btrbk/scripts (which the Makefile does). Doing what upstream does is the way to go.

**Comment by candrews** [comment]
@digint do you want to be included as a maintainer in metadata.xml?

**Comment by digint** [comment]
Yes you can add me as a maintainer.

Regarding the Makefile: are the systemd unit files copied to the correct place when using `emake install` rather than `systemd_dounit` ? If so, I agree that it's much cleaner this way.

You might also consider depending on `sys-apps/pv`. This is only used for the `--progress` option and `rate_limit` config option, but some people (on the debian package) complained about this missing dependency.

**Comment by candrews** [comment]
@digint,
> Yes you can add me as a maintainer.

Excellent, I've updated metadata.xml accordingly.
> Regarding the Makefile: are the systemd unit files copied to the correct place when using `emake install` rather than `systemd_dounit`? If so, I agree that it's much cleaner this way.

In both cases, the files are copied to `/usr/lib/systemd/system` which is probably almost always right. Thanks for point this out - I've changed the ebuilds to pass `SYSTEMDDIR="$(systemd_get_systemunitdir)"` to `emake` so that the path is always right :-)
> You might also consider depending on `sys-apps/pv`. This is only used for the `--progress` option and `rate_limit` config option, but some people (on the debian package) complained about this missing dependency.

I added a use flag to toggle this dependency - I think that approach makes sense.
> I'm releasing btrbk-0.23.3 later tonight (small bugfix, also adding missing ChangeLog), yell if there's something missing.

Thanks!

**Comment by digint** [comment]
@candrews thanks, I can confirm it working at my place.

**Comment by gktrk** [comment]
Merge in https://github.com/gentoo/gentoo/commit/932c16f63b5beb8f798ebf021a38fbbbee794072. Thanks!

I've made the following changes before merge:
- Removed extra empty line at the end of metadata.xml
- Mangled the commit message summary line slightly

**Pull request closed without merging**